src/main/java/pl/byczazagroda/trackexpensesappbackend/service/FinancialTransactionCategoryServiceImpl.java

```diff
@@ -0,0 +1,37 @@
+package pl.byczazagroda.trackexpensesappbackend.service;
+
+import lombok.RequiredArgsConstructor;
+import org.springframework.stereotype.Service;
+import pl.byczazagroda.trackexpensesappbackend.dto.FinancialTransactionCategoryDTO;
+import pl.byczazagroda.trackexpensesappbackend.dto.FinancialTransactionCategoryUpdateDTO;
+import pl.byczazagroda.trackexpensesappbackend.exception.AppRuntimeException;
+import pl.byczazagroda.trackexpensesappbackend.exception.ErrorCode;
+import pl.byczazagroda.trackexpensesappbackend.mapper.FinancialTransactionCategoryModelMapper;
+import pl.byczazagroda.trackexpensesappbackend.model.FinancialTransactionCategory;
+import pl.byczazagroda.trackexpensesappbackend.repository.FinancialTransactionCategoryRepository;
+
+import javax.transaction.Transactional;
+
+@Service
+@RequiredArgsConstructor
+public class FinancialTransactionCategoryServiceImpl implements FinancialTransactionCategoryService
+{
+    private final FinancialTransactionCategoryRepository financialTransactionCategoryRepository;
+    private final FinancialTransactionCategoryModelMapper financialTransactionCategoryModelMapper;
+
+
+    @Override
+    @Transactional
+    public FinancialTransactionCategoryDTO updateFinancialTransactionCategory(Long id, FinancialTransactionCategoryUpdateDTO financialTransactionCategoryUpdateDTO) {
+        FinancialTransactionCategory financialTransactionCategory
+                = financialTransactionCategoryRepository.findById(id)
+                .orElseThrow(() -> new AppRuntimeException(ErrorCode.FTC001,
+                        String.format("Financial transaction category with id: %d not found", id)));
+        financialTransactionCategory.setName(financialTransactionCategoryUpdateDTO.name());
+        financialTransactionCategory.setType(financialTransactionCategoryUpdateDTO.type());
+        FinancialTransactionCategory updated
+                = financialTransactionCategoryRepository.save(financialTransactionCategory);
+//        TODO check if category is currently assigned to any financial transaction then throw an error that u cannot update it
```

Comment:
OK.

Reply:
was mentioned in the isue that i need to check here if the FT Category is assigned to any Financial Transaction but these models are not related with each other in any way? I have read other issues about the model class implementation but they do not provide any information about the relationship beetwen these columns